src/partition/repart.c

```diff
@@ -5183,11 +5183,12 @@ static int context_open_copy_block_paths(
                                                        "Copying from block device node is not permitted in --image=/--root= mode, refusing.");
 
                 } else if (p->copy_blocks_auto) {
-                        dev_t devno;
+                        dev_t devno = 0;  /* Fake initialization to appease gcc. */
 
                         r = resolve_copy_blocks_auto(p->type, p->copy_blocks_root, restrict_devno, &devno, &uuid);
                         if (r < 0)
                                 return r;
+                        assert(devno != 0);
```

Comment:
my educated guess is is that gcc doesn't grok that in resolve_copy_blocks_auto() if opendir() returns NULL errno is definitely not 0. hence it assumes that we might end up returning with exit code zero but with devno not initialized, via the `if (errno != ENOENT)` codepath...

Reply:
I thought so too, but (as described in the commit message), it doesn't seem to be it. Adding `assert(errno > 0)` doesn't change things.